src/v/cluster/cluster_utils.h

```diff
@@ -25,6 +25,23 @@
 
 #include <utility>
 
+namespace detail {
+
+template<typename T, typename Fn>
+std::vector<cluster::topic_result>
+create_topic_results(const std::vector<T>& topics, Fn fn) {
+    std::vector<cluster::topic_result> results;
+    results.reserve(topics.size());
+    std::transform(
+      topics.cbegin(),
+      topics.cend(),
+      std::back_inserter(results),
+      [&fn](const T& t) { return fn(t); });
+    return results;
+}
```

Comment:
An alternative is to create a projection of the input to a `model::topic_namespace`, something like:

```cpp
struct project_tp_ns {
   const model::topic_namespace& operator()(const model::topic_namespace& tp_ns)  {
       return tp_ns;
   }
   const model::topic_namespace& operator()(const custom_assignable_topic_configuration& r) {
       return r.cfg.tp_ns
   }
   const model::topic_namespace& operator()(const non_replicable_topic& r) {
       return r.name;
   }
};

template<typename T>
std::vector<cluster::topic_result>
create_topic_results(const std::vector<T>& topics, errc error_code) {
    std::vector<cluster::topic_result> results;
    results.reserve(topics.size());
    std::transform(
      topics.cbegin(),
      topics.cend(),
      std::back_inserter(results),
      [error_code](const T& t) { return topic_result(project_tp_ns{}(t), ec; });
    return results;
```
That opens up the possibility of using a `boost::transform_iterator` with `std::copy`, but it's probably best not to pull in that dependency in the header, as I'm not sure it's any clearer.

Reply:
this is so cool

Reply:
:sadface: